packages/engine/src/FeatureFlagsHook.tsx

```diff
@@ -34,7 +34,7 @@ export function useFeatureFlags(flagNames: string[]): boolean[] {
     }
   })
 
-  if (!response.data) {
+  if (!response.data || response.status !== 'success') {
```

Comment:
data will be `[]` if it has not finished the request, so we can remove this whole conditional

Reply:
Yes, though we do need to make sure the feature is returned as disabled as the data is being fetched.

as discussed in https://tsu.atlassian.net/browse/IR-2888